data-pipeline/content-cache-updater/src/main/java/org/ekstep/ep/samza/service/ContentCacheUpdaterService.java

```diff
@@ -0,0 +1,153 @@
+            }
+        }
+
+    }
+
+    public Map<String, Object> updateDialCodeCache(Map<String, Object> message, ContentCacheUpdaterSink sink) {
+        String nodeUniqueId = (String) message.get("nodeUniqueId");
+        Map<String, Object> parsedData = null;
+            String contentNode = dialCodeStoreConnection.get(nodeUniqueId);
+            if (contentNode != null) {
+                parsedData = gson.fromJson(contentNode, mapType);
+            } else {
+                parsedData = new HashMap<>();
+            }
+            Map<String, Object> newProperties = extractProperties(message);
+            parsedData.putAll(newProperties);
+            if (parsedData.size() > 0) {
+                parsedData.values().removeAll(Collections.singleton(null));
+                parsedData.values().removeAll(Collections.singleton(""));
+            }
+        return parsedData;
+    }
+
+    @SuppressWarnings("unchecked")
+    public Map<String, Object> updateContentCache(Map<String, Object> message, ContentCacheUpdaterSink sink) {
+        String nodeUniqueId = (String) message.get("nodeUniqueId");
+        Map<String, Object> parsedData = null;
+        Map<String, List<String>> listTypeFields = new HashMap();
+            String contentNode = contentStoreConnection.get(nodeUniqueId);
+            if (contentNode == null) {
+                parsedData = new HashMap<>();
+            } else {
+                parsedData = gson.fromJson(contentNode, mapType);
+            }
+            Map<String, Object> newProperties = extractProperties(message);
+            for (Map.Entry<String, Object> entry : newProperties.entrySet()) {
+                if (contentModelListTypeFields.contains(entry.getKey())
+                        && entry.getValue() instanceof String
+                        && !((String) entry.getValue()).isEmpty()
+                ) {
+                    String str = (String) entry.getValue();
+                    List<String> value = toList(str);
+                    listTypeFields.put(entry.getKey(), value);
+                    LOGGER.info("", "type cast to List for field: " + entry.getKey() + " ,nodeUniqueId: " + nodeUniqueId);
+                }
+            }
+            if (!listTypeFields.isEmpty()) {
+                newProperties.putAll(listTypeFields);
+            }
+            parsedData.putAll(newProperties);
+            if (parsedData.size() > 0) {
+                parsedData.values().removeAll(Collections.singleton(null));
+                parsedData.values().removeAll(Collections.singleton(""));
+                redisConnect.addJsonToCache(nodeUniqueId, gson.toJson(parsedData), contentStoreConnection);
+                sink.success();
+            }
+        return parsedData;
+    }
+
+    private List<String> toList(String value) {
+        List<String> val = new ArrayList<>();
+        val.add(value);
+        return val;
+    }
```

Comment:
This method is unnecessary. We could have simply done Arrays.asList(value) or used other util methods